### PR TITLE
util/types: support convert year

### PR DESCRIPTION
**Pull request by coocood**
Convert support more types

### CODE REVIEW

File: util/types/convert.go [hunk 1]
```diff
@@ -138,6 +138,23 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		}
 		// TODO: check Flen
 		return x, nil
+	case mysql.TypeYear:
+		var intVal int64
```
Comment (qiuyesuifeng) [inline util/types/convert.go]: how about 
```
var (
         intVal int64
         err error
)
```

**Comment by siddontang** [comment]
There is no year support in mysql cast function? why to support this?

**Comment by coocood** [comment]
Convert be used to replace column.CastValue later.

File: util/types/convert.go [hunk 2]
```diff
@@ -138,6 +138,25 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
...
 		return x, nil
+	case mysql.TypeYear:
+		var (
+			intVal int64
+			err    error
+		)
+		switch x := val.(type) {
+		case string:
+			intVal, err = StrToInt(x)
+		default:
```
Comment (siddontang) [inline util/types/convert.go]: mysql> insert into t select cast("2015-11-11 11:11:11" as datetime);
Query OK, 1 row affected (0.01 sec)
Records: 1  Duplicates: 0  Warnings: 0

I think you should support time type separately.  ToInt64 will not work for time type.

File: util/types/convert.go [hunk 3]
```diff
@@ -138,6 +138,27 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
...
 		return x, nil
+	case mysql.TypeYear:
+		var (
+			intVal int64
+			err    error
+		)
+		switch x := val.(type) {
+		case string:
+			intVal, err = StrToInt(x)
+		case mysql.Time:
```
Comment (siddontang) [inline util/types/convert.go]: mysql.Duration will use current year.

**Pull request merged**